File: backend/internal/model/temperature_series.go
package model

import "time"

type TemperatureSeries struct {
	ID                uint        `gorm:"primaryKey" json:"id"`
	PourSectionID     uint        `gorm:"not null;index" json:"pour_section_id"`
	PourSection       PourSection `gorm:"foreignKey:PourSectionID" json:"pour_section,omitempty"`
	SensorCode        string      `gorm:"size:80;not null;index" json:"sensor_code"`
	SampleIntervalMin int         `gorm:"not null" json:"sample_interval_min"`
	PointsJSON        string      `gorm:"type:text;not null" json:"points_json"`
	StartedAt         time.Time   `gorm:"not null" json:"started_at"`
	EndedAt           time.Time   `gorm:"not null" json:"ended_at"`
	SourceChecksum    string      `gorm:"size:64;not null;uniqueIndex" json:"source_checksum"`
	MissingRatio      float64     `gorm:"not null" json:"missing_ratio"`
	SeriesState       string      `gorm:"size:24;not null;index" json:"series_state"`
	QualityNote       string      `gorm:"type:text;not null" json:"quality_note"`
	ImportedBy        uint        `gorm:"not null;index" json:"imported_by"`
	ImportedByName    string      `gorm:"size:120;not null" json:"imported_by_name"`
	CreatedAt         time.Time   `json:"created_at"`
	UpdatedAt         time.Time   `json:"updated_at"`
}

File: frontend/src/types/temperature-series.ts
export type TemperatureSeriesState = 'imported' | 'usable' | 'invalid'

export interface TemperaturePoint {
  timestamp: string
  temperature_c: number
}

export interface TemperatureSeries {
  id: number
  pour_section_id: number
  section_code: string
  sensor_code: string
  sample_interval_min: number
  points: TemperaturePoint[]
  started_at: string
  ended_at: string
  source_checksum: string
  missing_ratio: number
  series_state: TemperatureSeriesState
  quality_note: string
  imported_by: number
  imported_by_name: string
  created_at: string
  updated_at: string
}

export interface ImportTemperatureSeries {
  pour_section_id: number
  sensor_code: string
  sample_interval_min: number
  points: TemperaturePoint[]
  quality_note?: string
}
